notebooks/smsnet_ensemble.ipynb

```diff
@@ -0,0 +1,493 @@
+   "outputs": [],
+   "source": [
+    "from    src.network         import  build_network\n",
+    "from    src.parameters      import  MSDNetParameters, TUNetParameters, TUNet3PlusParameters\n",
+    "from    src.seg_utils       import  train_val_split, train_segmentation\n",
+    "from    src.tiled_dataset   import  TiledDataset\n",
+    "import  torch\n",
+    "import  torch.nn        as      nn\n",
+    "import  torch.optim     as      optim\n",
+    "from    torchvision     import  transforms\n",
+    "from    src.utils           import  create_directory\n",
+    "\n",
+    "import torch\n",
+    "from torch.utils.data import DataLoader, random_split\n",
+    "from torch.utils.data.dataloader import default_collate\n",
+    "from dlsia.core.train_scripts import segmentation_metrics\n",
+    "import yaml"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Open the YAML file for all parameters\n",
+    "yaml_path = 'example_yamls/example_smsnet_ensemble.yaml'\n",
+    "with open(yaml_path, 'r') as file:\n",
+    "    # Load parameters\n",
+    "    parameters = yaml.safe_load(file)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "model_parameters = parameters['model_parameters']\n",
+    "model_parameters"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset = TiledDataset(\n",
+    "        data_tiled_uri=parameters['data_tiled_uri'],\n",
+    "        data_tiled_api_key=parameters['data_tiled_api_key'],\n",
+    "        mask_tiled_uri=parameters['mask_tiled_uri'],\n",
+    "        mask_tiled_api_key=parameters['mask_tiled_api_key'],\n",
+    "        qlty_window=model_parameters['qlty_window'],\n",
+    "        qlty_step=model_parameters['qlty_step'],\n",
+    "        qlty_border=model_parameters['qlty_border'],\n",
+    "        transform=transforms.ToTensor()\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset.mask_client"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset.mask_client_one_up.metadata"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "for n in range(3):\n",
+    "    plt.imshow(dataset[n])\n",
+    "    plt.colorbar()\n",
+    "    plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset.mask_client"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def custom_collate(batch):\n",
```

Comment:
Is this function different from the one defined in `seg_utils.py`?

Reply:
Most of the SMSNet stuff is similar to other models. 

FYI the implementation of SMSNet Ensemble has been ingested into src scripts. You can safely refer to those and discard the notebook.